lib/user-interface/react/src/shared/model/model-management.model.ts

```diff
@@ -29,11 +31,142 @@ export enum ModelType {
     embedding = 'embedding',
 }
 
+export enum InferenceContainer {
+    TGI = 'tgi',
+    TEI = 'tei',
+    VLLM = 'vllm',
+    INSTRUCTOR = 'instructor',
+}
+
+export type IContainerHealthCheckConfig = {
+    Command: string[];
+    Interval: number;
+    StartPeriod: number;
+    Timeout: number;
+    Retries: number;
+};
+
+export type IContainerConfigImage = {
+    BaseImage: string;
+    Path: string;
+    Type: string;
+};
+
+export type IMetricConfig = {
+    AlbMetricName: string;
+    TargetValue: number;
+    Duration: number;
+    EstimatedInstanceWarmup: number;
+};
+
+export type ILoadBalancerHealthCheckConfig = {
+    Path: string;
+    Interval: number;
+    Timeout: number;
+    HealthyThresholdCount: number;
+    UnhealthyThresholdCount: number;
+};
+
+export type ILoadBalancerConfig = {
+    HealthCheckConfig: ILoadBalancerHealthCheckConfig
+};
+
+export type IAutoScalingConfig = {
+    MinCapacity: number;
+    MaxCapacity: number;
+    Cooldown: number;
+    DefaultInstanceWarmup: number;
+    MetricConfig: IMetricConfig;
+};
+
+export type IContainerConfig = {
+    BaseImage: IContainerConfigImage;
+    SharedMemorySize: number;
+    HealthCheckConfig: IContainerHealthCheckConfig;
+    Environment?: Record<string, string>[];
+};
+
 export type IModel = {
-    ModelName: string;
     ModelId: string;
+    ModelName: string;
+    Streaming: boolean;
     ModelType: ModelType;
-    ModelUrl: string;
+    InstanceType: string;
+    InferenceContainer: InferenceContainer;
+    ContainerConfig: IContainerConfig;
+    AutoScalingConfig: IAutoScalingConfig;
+    LoadBalancerConfig: ILoadBalancerConfig;
+};
+
+export type IModelRequest = {
```

Comment:
No action to take, but also to note (primarily for myself), this will need to have the ModelUrl added as a potential field as well. In the case that we add something directly to LiteLLM and don't create the infrastructure for it, we'd need the ability to set the model URI too.